specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,273 @@
+                            // , we can parse it and get the detail what we need.
+                            wil::unique_cotaskmem_string exceptionDetail;
+                            if (exception && exception->get_Detail(&exceptionDetail) == S_OK)
+                            {
+                                MessageBox(
+                                    nullptr, exceptionDetail.get(), L"ExecuteScript Result", MB_OK);
+                            }
+                        }
+                        else
+                        {
+                            MessageBox(
+                                nullptr, L"Get exception failed", L"ExecuteScript Result", MB_OK); 
+                        }
+                    }
+
+                }
+            }
+        )
+    )
+}
+
+```
+## .NET and WinRT
+
+```c#
+void ExecuteScriptWithResultAsync(String script)
+{
+    var environment = webView2Control.CoreWebView2.Environment;
+    CoreWebView2ExecuteScriptResult result = await ExecuteScriptWithResultAsync(script);
+
+    bool isSuccess = result.IsSuccess;
+    if (isSuccess) 
+    {
+        Debug.WriteLine($"execute script success.");
+
+        Debug.WriteLine($"json result received: {result.ResultAsJson}");
+        try 
+        {
+            string stringResult = result.TryGetResultAsString();
+            Debug.WriteLine($"get string result success: {stringResult}");
+        } 
+        catch (ArgumentException) 
+        {
+            Debug.WriteLine($"Non-string message received");
+        }
+    } 
+    else 
+    {
+        Debug.WriteLine($"execute script failed.");
+        CoreWebView2ExecuteScriptException exception = result.Exception;
+        Debug.WriteLine($"exception name: {exception.Name}");
+        Debug.WriteLine($"exception message: {exception.Message}");
+        Debug.WriteLine($"exception Detail: {exception.Detail}");
+    }
+}
+```
+
+# API Details
+## Win32 C++
+```c++
+/// This interface represents a JavaScript exception. If the CoreWebView2.ExecuteScriptWithResult result has IsSuccessful as false, you can use the result's Exception property to get the script exception.
+/// use get_Exception to get it.
+[uuid(82F22B72-1B22-403E-A0B9-A8816C9C8E45), object, pointer_default(unique)]
+interface ICoreWebView2ScriptException : IUnknown {
+    /// The line number of the source where the exception occurred.
+    [propget] HRESULT LineNumber([out, retval] UINT* value);
+    /// The column number of the source where the exception occurred.
+    [propget] HRESULT ColumnNumber([out, retval] UINT* value);
+
+  /// The Name is the exception's class name. In the JSON it is `exceptionDetail.exception.className`. This is the empty string if the exception doesn't have a class name. This can happen if the script throws a non-Error object such as `throw "abc";`
+  /// `result.exceptionDetail.exception.className` in json result, this
```

Comment:
This line 174, 175, and 176 can be removed since they are redundant with line 173